app/atoms/records.ts

```diff
@@ -194,25 +287,83 @@ export const recordAttributeAtom = atom(
       // Populate this record attribute on mount
       populate();
 
+      const cleanupListener = intervalInstance.on("change", ({ newState }) => {
+        signal.mutate(({ value }) => ({
+          value: value ? `${value.slice(0, -2)} ${newState}` : value,
+        }));
+      });
+
       return () => {
         // Cancel the population - this is relevant if the record attribute is mounted and unmounted quickly,
         // for example when scrolling through a table
+        /**
+         * @NOTE this will only happen if this atom has `ttl` or is manually
+         * destroyed. I've added `ttl: 0`
+         */
         cancelPopulate();
+        cleanupListener();
       };
-    }, []);
+    }, [intervalInstance]);
```

Comment:
Does the `intervalInstance` dep here, given that the `intervalAtom` takes no args and so should always be the same? Should `populate` be here as well even though it should theoretically never change since it's a`injectCallback` w no args?

Reply:
`populate` definitely doesn't need to be here. I would put `intervalInstance` here. I think I meant to leave a note about it.

Atom instances aren't technically stable references. They can be force-destroyed. The ecosystem can also be `.reset()`, which force-destroys all atom instances, recreating them from the leaves (typically React components). If the `intervalAtom`'s single instance is force-destroyed, this atom would reevaluate, creating a new `intervalInstance`.

But do with those features what you will. We actually never use them, so in our apps we do consider this a stable reference. I kind of wish we didn't do that though. In official docs and examples, I regard atom instances as unstable references.

Reply:
`injectMemo` has that (in v2)!

```ts
injectMemo(() => {
  otherAtom.get() // tracked
  someSignal.get() // tracked
}) // don't pass a deps array
```

It doesn't make sense for `injectCallback` to have this overload - a) deps wouldn't be tracked until it's called; and b) callbacks are typically used in exports, so need to be stable anyway.

As for `injectEffect`, as SolidJS is experiencing right now, auto-tracking async effect dependencies is annoying. It won't work without generators + weird APIs or the Async Context proposal. If async context ever reaches adoption, I will consider adding this overload to `injectEffect`.